File: Leetcode/2 - Add Two Numbers/solution.py
def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
    resList = ListNode(0)
    current = resList
    carry = 0

    while l1 is not None or l2 is not None or carry > 0:
        val1 = l1.val if l1 else 0
        val2 = l2.val if l2 else 0

        sum_ = val1 + val2 + carry
        carry = sum_ // 10

        current.next = ListNode(sum_ % 10)
        current = current.next

        if l1:
            l1 = l1.next
        if l2:
            l2 = l2.next

    return resList.next

File: Leetcode/2 - Add Two Numbers/solution.js
var addTwoNumbers = function (l1, l2) {
    let resList = new ListNode(0);
    let current = resList;
    let carry = 0;

    while (l1 !== null || l2 !== null || carry > 0) {
        let val1 = l1 ? l1.val : 0;
        let val2 = l2 ? l2.val : 0;

        let sum = val1 + val2 + carry;
        carry = Math.floor(sum / 10);

        current.next = new ListNode(sum % 10);
        current = current.next;

        if (l1) l1 = l1.next;
        if (l2) l2 = l2.next;
    }

    return resList.next;
};